### DISCUSSION TITLE
Frameworks for Tauri

### DISCUSSION THREAD

**Discussion by TheLexoPlexx** [tauri-apps/tauri#7470] [General]
Hi there,

I just got into tauri as a side-project because I wanted more Rust in my life. I simply ported my nextjs13-Project to tauri and it worked quite well so far. I needed to seperate client and server of course which is why I set up a simple Websocket-TRPC-Server and wanted to connect to that.

The server works just fine. But the client doesn't. Due to nextjs rerendering the page "server-side" first, `window` is `undefined` and I can't really disable SSR, there's a package (incompatible with the current React-Version) and settings like `dynamic()` but none of them seem to do the trick and by now, almost all the files have a `"use client"`-directive at the top.

So clearly, Nextjs is kinda cool, but at the current state of turbopack and appDir not very well suited. (It's react as well and I wanted less react in the future).

I'll see if I can fix the problem anyways (or maybe just ditch the websocket) but I'm also trying to explore other Frameworks. I guess none of them are made for client-only.

What about Svelte(Kit)? Or maybe even full-on-Rust with Leptos or Yew?

Personally, I'm looking for the ability to create a typesafe (trpc-like) backend running prisma (and/or a typesafe query-builder) and a tauri-app with multiple client including authentication. Most of which can be done through trpc and trpc-shield afaik.

Or maybe, just maybe, I'm completelly off the track here and someone needs to get me back to the road.

Looking forward to an exciting discussion.

**Comment by Manubi** [comment]
I als would be interested in getting some valuable feedback regarding nextjs and/or leptos. 
The challenge I see is, many want to use nextjs for the web which makes sense but will probably be a pain for tauri?  I also would be interested in getting to know what's the plan with tauri 2.0 and the mobile strategy for this.
Basically what I want to know, what are the possible strategies for tauri with a web/desktop/mobile application. Thanks

**Reply by TheLexoPlexx** [reply]
Yes, my experience was that due to SSR and other Server-Side-Stuff Nextjs was a pain to use with tauri and I ultimately had to ditch tauri again and take a different approach, even though I don't like it.

It was also not possible to properly turn off all the Server-Side-Stuff in netxtjs.

**Reply by Manubi** [reply]
Didn't you consider using react/vite with tauri? that should work pretty fine?

**Reply by TheLexoPlexx** [reply]
I did not, so far. I thought about trying Svelte as well but I started with Nextjs and the app had already grown quite a lot so it's either time for a larger rewrite or I had to find other solutions and try either of those (Leptos, Svelte, Vite) in the next project.

**Comment by dukedorje** [comment]
I've been using SvelteKit with SSG-mode with Tauri to great success.  Still trying to figure out a way to generate full type-safe Typescript calls like tRPC without having to open up a new socket.
